inspire/base/templates/records/Inspire_Default_HTML_detailed_macros.tpl

```diff
@@ -0,0 +1,176 @@
+{#
+# This file is part of INSPIRE.
+# Copyright (C) 2015 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#}
+
+{% macro display_collections() %}
+  {% if record.get('collections') %}
+  {% set collection_found = [] %}
+    {% set collections = record.collections %}
+    {% for collection in collections %}
+      {% set collection_name = collection.get('primary') %}
+      {% if collection_name == "HEP" %}
+        {% if not collection_found %}
+          <span class="currect-collection">Literature</span>
+          <span>Authors</span>
+          <span>Conferences</span>
+          <span>Jobs</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% elif collection_name == "HEPNAMES" and collection_found %}
+        {% if not collection_found %}
+          <span>Literature</span>
+          <span class="currect-collection">Authors</span>
+          <span>Conferences</span>
+          <span>Jobs</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% elif collection_name == "CONFERENCES" and collection_found %}
+        {% if not collection_found %}
+          <span>Literature</span>
+          <span>Authors</span>
+          <span class="currect-collection">Conferences</span>
+          <span>Jobs</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% elif collection_name == "JOB" and collection_found %}
+        {% if not collection_found %}
+          <span>Literature</span>
+          <span>Authors</span>
+          <span>Conferences</span>
+          <span class="currect-collection">Jobs</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% endif %}
+    {% endfor %}
+    {% if not collection_found %}
+      <span>Literature</span>
+      <span>Authors</span>
+      <span>Conferences</span>
+      <span>Jobs</span>
+    {% endif %}
+  {% endif %}
```

Comment:
This code can probably be heavily simplified as there is a lot of repetition just to add a class.

Isn't it enough with something like this for each collection?
```
{% if collection_name == "HEP" %}
<span class="currect-collection">Literature</span>
{% else %}
<span>Literature</span>
{% endif %}
```

There might be even be better ways using, for example, string formatting http://jinja.pocoo.org/docs/dev/templates/#format